lib/plugins/aws/deploy/lib/checkForChanges.js

```diff
@@ -207,113 +207,101 @@ module.exports = {
   async checkLogGroupSubscriptionFilterResourceLimitExceeded() {
     const region = this.provider.getRegion();
 
-    return this.provider.getAccountInfo().then((account) =>
-      Promise.all(
-        this.serverless.service.getAllFunctions().map((functionName) => {
-          const functionObj = this.serverless.service.getFunction(functionName);
-
-          if (!functionObj.events) {
-            return BbPromise.resolve();
-          }
-
-          let logSubscriptionSerialNumber = 0;
-          const promises = functionObj.events.map((event) => {
-            if (!event.cloudwatchLog) {
-              return BbPromise.resolve();
-            }
-
-            let logGroupName;
-
-            /*
-              it isn't necessary to run sanity checks here as they already happened during the
-              compile step
-            */
-            if (typeof event.cloudwatchLog === 'object') {
-              logGroupName = event.cloudwatchLog.logGroup.replace(/\r?\n/g, '');
-            } else {
-              logGroupName = event.cloudwatchLog.replace(/\r?\n/g, '');
-            }
-
-            const accountId = account.accountId;
-            const partition = account.partition;
+    const account = await this.provider.getAccountInfo();
+    const accountId = account.accountId;
+    const partition = account.partition;
+
+    const functionNames = await this.serverless.service.getAllFunctions();
+    const cloudwatchLogEvents = _.flatten(
+      functionNames.map((functionName) => {
+        const functionObj = this.serverless.service.getFunction(functionName);
+        const FunctionName = functionObj.name;
+        const events = functionObj.events;
+        let logSubscriptionSerialNumber = 0;
+        return events
+          .filter((event) => !!event.cloudwatchLog)
+          .map((event) => {
+            const logGroupName = event.cloudwatchLog.logGroup || event.cloudwatchLog;
```

Comment:
Is there a reason as to why the `replace` logic is no longer applied to `logGroupName`? I'm guessing that's because it's already applied on schema level but I want to be sure as it's a change from previous behavior

Reply:
I thought like what you said -its already applied on schema level-,
but it seems like even if its applied on compile time, we should replace again at `deploy:checkForChanges` since the replacements didnt effects to the origin event 😢 

I will re-apply the replacements. thanks for checking 👍 👍